Analytical_Engine/Modify/SetOutlineElements1D.cs

```diff
@@ -70,6 +70,19 @@ public static IOpening<TEdge> SetOutlineElements1D<TEdge>(this IOpening<TEdge> o
             return pp;
         }
 
+        /***************************************************/
+
+        [Description("Sets the Outline Element1Ds of an IRegion, i.e. the perimiter. Method required for all IElement2Ds.")]
+        [Input("region", "The IRegion to update the Perimeter of.")]
+        [Input("outlineElements", "A list of IElement1Ds which all should be Geometrical ICurves.")]
+        [Output("region", "The region with updated perimiter.")]
+        public static IRegion SetOutlineElements1D(this IRegion region, IEnumerable<IElement1D> outlineElements)
+        {
+            IRegion r = region.GetShallowClone(true) as IRegion;
+            r.Perimeter = BH.Engine.Geometry.Compute.IJoin(outlineElements.Cast<ICurve>().ToList())[0];
```

Comment:
I would consider adding a check if the output of `IJoin` contains only 1 item and if that item is closed - if not, raise a warning that the output may be wrong.